src/Tools/dotnet-monitor/Egress/AzureBlob/AzureBlobEgressProvider.cs

```diff
@@ -129,6 +161,83 @@ private Uri GetAccountUri(AzureBlobEgressProviderOptions options, out string acc
             return blobUriBuilder.ToUri();
         }
 
+        private Uri GetQueueAccountUri(AzureBlobEgressProviderOptions options, out string accountName)
+        {
+            var queueUriBuilder = new QueueUriBuilder(options.QueueAccountUri);
+
+            queueUriBuilder.Query = null;
+            queueUriBuilder.QueueName = null;
+
+            accountName = queueUriBuilder.AccountName;
+
+            return queueUriBuilder.ToUri();
+        }
+
+        private async Task EgressMessageToQueue(string blobName, AzureBlobEgressProviderOptions options, CancellationToken token)
+        {
+            try
+            {
+                QueueClient queueClient = await GetQueueClientAsync(options, token);
+
+                if (queueClient.Exists())
+                {
+                    queueClient.SendMessage(Convert.ToBase64String(Encoding.UTF8.GetBytes(blobName)));
```

Comment:
Previously, I was using `QueueClientOptions` to handle the Base64 conversion; however, after switching to an older version of Azure Queues, the message encoding option doesn't appear to be available. This should have the same effect.

Reply:
Why does it need to be base64 encoded before calling SendMessage? Shouldn't we just be able to give it whatever is our raw data and it will encode in whatever format their protocol requires? This seems like the consumer would have to understand that the data needs to be base64 decoded, which I would not expect.

Reply:
Originally, I wasn't doing the encoding, and noticed that all of my messages were going into a queue named `dotnet-monitor-testing-queue-poison` - after looking around online and trying that, it resolved the issue. The error I was getting was "The input is not a valid Base-64 string as it contains a non-base 64 character, more than two padding characters, or an illegal character among the padding characters azure function queue message."

Reply:
How about: `await queueClient.SendMessageAsync(BinaryData.FromString(blobName), cancellationToken: token);`?

Reply:
Mostly redundant, but do please use the async method.